fe/fe-core/src/main/java/org/apache/doris/alter/CloudSchemaChangeJobV2.java

```diff
@@ -106,14 +110,44 @@ protected void commitShadowIndex() throws AlterCancelException {
     }
 
     @Override
-    protected void postProcessShadowIndex() {
+    protected void onCancel() {
         if (Config.enable_check_compatibility_mode) {
             LOG.info("skip drop shadown indexes in checking compatibility mode");
             return;
         }
 
         List<Long> shadowIdxList = indexIdMap.keySet().stream().collect(Collectors.toList());
         dropIndex(shadowIdxList);
+
+        long tryTimes = 1;
+        while (true) {
+            try {
+                Set<Table.Cell<Long, Long, Map<Long, Long>>> tableSet = partitionIndexTabletMap.cellSet();
+                Iterator<Cell<Long, Long, Map<Long, Long>>> it = tableSet.iterator();
+                while (it.hasNext()) {
+                    Table.Cell<Long, Long, Map<Long, Long>> data = it.next();
+                    Long partitionId = data.getRowKey();
+                    Long shadowIndexId = data.getColumnKey();
+                    Long originIndexId = indexIdMap.get(shadowIndexId);
+                    Map<Long, Long> shadowTabletIdToOriginTabletId = data.getValue();
+                    for (Map.Entry<Long, Long> entry : shadowTabletIdToOriginTabletId.entrySet()) {
```

Comment:
try best to cancel sc job even there is a RPC error

Reply:
```
// fe/fe-core/src/main/java/org/apache/doris/cloud/datasource/CloudInternalCatalog.java:removeSchemaChangeJob
int tryTimes = 0;
        while (tryTimes++ < Config.metaServiceRpcRetryTimes()) {
            try {
                response = MetaServiceProxy.getInstance().finishTabletJob(request);
                if (response.getStatus().getCode() != Cloud.MetaServiceCode.KV_TXN_CONFLICT) {
                    break;
                }
            } catch (RpcException e) {
                LOG.warn("tryTimes:{}, finishTabletJob RpcException", tryTimes, e);
                if (tryTimes + 1 >= Config.metaServiceRpcRetryTimes()) {
                    throw new DdlException(e.getMessage());
                }
            }
            sleepSeveralMs();
        }
```